### DISCUSSION TITLE
Obtaining Historical Data for Delisted/Defunct Tickers

### DISCUSSION THREAD

**Discussion by DanielMcNultyII** [alvarobartt/investpy#396] [Q&A]
First off, I think this package is awesome!

That said, I'm trying to compile data for S&P 500 tickers from about 1970 to 2021, but there are a number of tickers which have been delisted and/or are now defunct. Is there a way to get this information in Python using this package if investing.com has it?

For example, one of these tickers is Linear Technology (LLTC) which was delisted from the S&P 500 and bought by another company in 2017. When I search for the ticker LLTC using the code in your read-me for AAPL, I receive an error. However, the data is on investing.com here: [https://www.investing.com/equities/linear-technology-corp-historical-data](https://www.investing.com/equities/linear-technology-corp-historical-data). That said, investing.com itself doesn't return this when searching "LLTC" in it's own search bar and the only way I found it was through an external Google search.

Thank you for reading through this and for your time. I hope this doesn't come across as a complaint, because I really like this package. I'm just curious about this peculiar case.

**Comment by datatalking** [comment]
So you are looking for stock market data for old tickers 1970 to 2021, I might have a way to source for that.

I'm newer to investpy, once people have the data what platform options are available to semi or fully automate the trading?

**Reply by DanielMcNultyII** [reply]
Yes, that is what I am looking for. I thought it could be possible to get it through scraping Wikipedia for the list of companies comprising the S&P 500 each year then using investpy to pull the data, but then I ran into this peculiar case.

I don't have an answer for a platform to automate your trading though.
